app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+
+  has_many :votes
+
+  validates :name, presence: true
+end
+
+# QUESTION: I tried stripping white space from the beginning and end of the User name that was entered, but either it passed my tests but didn't work on my rails app, or it broke the test that a vote could be created with all fields. below are two examples of what I tried:
+
+  # 1.
+    # before_validation :strip_blanks
+    # def strip_blanks
+    #   self.name = self.name.strip!
+    # end
```

Comment:
You were really close here! A model filter is definitely the way to go, the disconnect is in the return value of `String#strip!`.

`String#strip` (no `!`) creates a new string and returns it. `String#strip!` modifies the original string, but unlike `strip` it returns `nil` if the string wasn't modified.

So, you should either call `strip` and set `self.name` to the result, or call `strip!`, but not both. In other words:

```ruby
self.name = self.name.strip
# - OR -
self.name.strip!
```

[Here is the ruby doc describing this behavior](https://ruby-doc.org/core-2.4.2/String.html#method-i-strip-21) which, as far as I'm concerned, is really unintuitive. You're not the first student to get bitten by this.

Reply:
I'm not familiar with `normaize_attribute`, is that from a separate gem?